### PR TITLE
Make various updates to Java Card configuration in prep for production flows

### PR DESCRIPTION
**Pull request by arsalansufi**
_Easiest reviewed by commit_

## Overview

Issue link: https://github.com/votingworks/vxsuite/issues/3169

A very first step as I work on production machine/card configuration flows, this PR makes various updates to Java Card configuration. It:

- Removes the openssl config file as a config param, since it's the same across all envs
- Adds dev vs. prod branching
- Uses `__dirname` to avoid having to pass `pathToAuthLibRoot`

## Testing

- [x] Updated unit tests
- [x] Manually tested that dev still works
- [x] Manually tested that prod will hypothetically work (by simulating aspects of prod in my dev environment)

## Checklist

- [ ] ~I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced~ N/A
- [ ] ~I have added JSDoc comments to any newly introduced exports~ N/A

### CODE REVIEW

File: libs/auth/src/java_card_config.ts [hunk 1]
```diff
@@ -15,36 +18,62 @@ export interface JavaCardConfig {
...
+      __dirname,
+      '../certs/dev/vx-admin-cert-authority-cert.pem'
+    ),
     vxAdminPrivateKeyPassword: DEV_PRIVATE_KEY_PASSWORD,
-    vxAdminPrivateKeyPath: `${pathToAuthLibRoot}/certs/dev/vx-admin-private-key.pem`,
+    vxAdminPrivateKeyPath: path.join(
+      __dirname,
+      '../certs/dev/vx-admin-private-key.pem'
+    ),
   };
```
Comment (arsalansufi) [inline libs/auth/src/java_card_config.ts:60]: As you can sort of see here, my plan for prod is to have `vxsuite-complete-system` machine configuration scripts create and store the VxAdmin cert authority cert at `/vx/config/vx-admin-cert-authority-cert.pem`. Until we're using the TPM for signing, the scripts can also store the VxAdmin private key at `/vx/config/vx-admin-private-key.pem` and provide the private key password through the same env var mechanism that we use to provide [machine ID](https://github.com/votingworks/vxsuite-complete-system/blob/7226af99dfc1a552036f35e2e769fd95786abb93/config/read-vx-machine-config.sh#L36-L39).

Dev (including VxDev) will continue to use the dev keys and certs committed to the `certs/dev` directory.

File: libs/auth/certs/prod/vx-cert-authority-cert.pem [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# A placeholder (and copy of certs/dev/vx-cert-authority-cert.pem)
+# until we have an official cert
```
Comment (arsalansufi) [inline libs/auth/certs/prod/vx-cert-authority-cert.pem:2]: TIL re PEM file comments - openssl knows to ignore anything outside the `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----`

File: libs/utils/src/environment_variable.ts [hunk 3]
```diff
@@ -124,7 +124,7 @@ export function getBooleanEnvVarConfig(
     case BooleanEnvironmentVariableName.ENABLE_JAVA_CARDS:
       return {
         name,
-        allowInProduction: false,
+        allowInProduction: true,
```
Comment (arsalansufi) [inline libs/utils/src/environment_variable.ts:127]: NH pilot, here we come!

File: libs/auth/src/java_card_config.ts [hunk 4]
```diff
@@ -15,36 +18,62 @@ export interface JavaCardConfig {
...
 /**
  * The password for all dev private keys
  */
 export const DEV_PRIVATE_KEY_PASSWORD = '1234';
 
-function constructDevCardProgrammingConfig(
-  pathToAuthLibRoot: string
-): JavaCardConfig['cardProgrammingConfig'] {
+function shouldUseProdCerts(): boolean {
+  return process.env.NODE_ENV === 'production' && !isVxDev();
```
[the remark this answers is not in the document]
Comment (arsalansufi) [reply libs/auth/src/java_card_config.ts:32]: Come next week, I'll either add a feature flag to enable usage of dev certs in prod or will have enough of the production flow ready that this won't be necessary

File: libs/auth/src/java_card_config.ts [hunk 5]
```diff
@@ -15,36 +18,62 @@ export interface JavaCardConfig {
   vxCertAuthorityCertPath: string;
 }
 
+/**
+ * Should be kept in sync with vxsuite-complete-system scripts
+ */
+const PROD_CONFIG_DIRECTORY = '/vx/config';
```
Comment (benadida) [inline libs/auth/src/java_card_config.ts:24]: I'd love for this to be an environment variable for configuration. Not required in this PR, but just food for thought.

**Review by benadida — APPROVED** [review]

File: libs/auth/src/java_card_config.ts (hunk 5 above)
Comment (arsalansufi) [reply libs/auth/src/java_card_config.ts:24]: Ah nice, I like that - will make it such that we don't have to keep vxsuite in sync with vxsuite-complete-system. It'll just be vxsuite-complete-system's responsibility to be internally consistent

Comment (arsalansufi) [reply libs/auth/src/java_card_config.ts]: https://github.com/votingworks/vxsuite/pull/3226/commits/55aaca6ebb915c34f5944cb9fd8c28b0a6e48d6a

**Pull request merged**